src/index.ts

```diff
@@ -3,14 +3,25 @@ import binify, {Command} from '@pnpm/package-bins'
 import {fromDir as readPackageJsonFromDir} from '@pnpm/read-package-json'
 import {PackageJson} from '@pnpm/types'
 import cmdShim = require('@zkochan/cmd-shim')
+import isWindows = require('is-windows')
 import mkdirp = require('mkdirp-promise')
 import Module = require('module')
 import fs = require('mz/fs')
 import normalizePath = require('normalize-path')
 import path = require('path')
+import pathKey = require('path-key')
 import R = require('ramda')
 import getPkgDirs from './getPkgDirs'
 
+const POWER_SHELL_IS_SUPPORTED = isPowerShellSupported()
+
+function isPowerShellSupported () {
+  if (isWindows()) return true
+
+  const pathValue = process.env[pathKey()]
+  return pathValue && pathValue.indexOf('pwsh') !== -1
```

Comment:
This likely won’t work, as `PATH` contains the directories in which `pwsh` can be located, eg. `/bin:/usr/bin:…`, and `pwsh` ends up in one of those directories.